test/RibbonOptionsVault.js

```diff
@@ -740,15 +749,26 @@ function behavesLikeRibbonOptionsVault(params) {
     describe("#deposit", () => {
       time.revertToSnapshotAfterEach();
 
-      it("deposits successfully", async function () {
-        const depositAmount = parseEther("1");
-
+      beforeEach(async function () {
         // Deposit only if asset is WETH
         if (params.asset === WETH_ADDRESS) {
-          await this.assetContract
-            .connect(userSigner)
-            .deposit({ value: depositAmount });
+          const addressToDeposit = [
+            userSigner,
+            managerSigner,
+            counterpartySigner,
+            adminSigner,
+          ];
+
+          for (let i = 0; i < addressToDeposit.length; i++) {
+            const weth = this.assetContract.connect(addressToDeposit[i]);
+            await weth.deposit({ value: parseEther("10") });
+            await weth.approve(this.vault.address, parseEther("10"));
+          }
```

Comment:
Same refers to this https://github.com/ribbon-finance/structured-products/pull/115/commits/16164ad8f14f92afe87d74d4faa0c7b87123d98e#r613789516